### DISCUSSION TITLE
[Bug] met a  interface conversion panic of skywalking.http.flush problem

### DISCUSSION THREAD

**Discussion by akilichen** [apache/skywalking#12617] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Go Agent (apache/skywalking-go)

### What happened

I am using the standard package named `httputil` with its `NewSingleHostReverseProxy` function, here is the error messages encountered.
```
panic: interface conversion: *mux.skywalking_operatorTypeMuxwriterWrapper is not http.Flusher: missing method Flush

goroutine 62 [running]:
github.com/gin-gonic/gin.(*responseWriter).Flush(0xc0001a2500?)
        /app/go-project/ms-gateway/vendor/github.com/gin-gonic/gin/response_writer.go:118 +0x59
net/http/httputil.(*maxLatencyWriter).delayedFlush(0xc0006939b0)
        /usr/local/go/src/net/http/httputil/reverseproxy.go:546 +0x95
created by time.goFunc
        /usr/local/go/src/time/sleep.go:176 +0x32
```
here is my code.
```
        proxy := httputil.NewSingleHostReverseProxy(remote) // remote mean target host.
	//Define the director func
	urlPath := c.Param("proxyPath")
	proxy.Director = func(req *orginHttp.Request) {
		req.Header = c.Request.Header
		req.Host = remote.Host
		req.URL.Scheme = remote.Scheme
		req.URL.Host = remote.Host
		req.URL.Path = c.Request.URL.Path
		// some code operation here
		......
	}
	proxy.ServeHTTP(c.Writer, c.Request)
```
something else: 
the sw-agent version is `v0.3.0` with `go1.19`.

### What you expected to happen

fix the bug!
Is the httputil reverse proxy not supported by SW, or something I did wrong? What should I do?

### How to reproduce

see `What happened`

### Anything else

If anything I do not explain clearly, leave me a message. Thanks!

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Please at least use the latest go-agent release. 0.3 is months ago.

**Reply by akilichen** [reply]
We originally planned not to use the latest go-agent release, because it needs to upgrade most of the package's dependencies, which may lead to some unexpected error. So it's nothing to do but use the latest go-agent release?

**Reply by wu-sheng** [reply]
Please read the change log. Many changes are included from 0.3 to the latest.

> We originally planned not to use the latest go-agent release

That is your call, but for the community, we only maintain the latest.

**Reply by akilichen** [reply]
Thanks for your answers, I will use the latest release first!

**Reply by akilichen** [reply]
I tried the latest release v0.5.0, but the bug is still. Here is the new error messages:
```
panic: interface conversion: *mux.skywalking_operatorTypeMuxwriterWrapperWithHijacker is not http.Flusher: missing method Flush

goroutine 60 [running]:
github.com/gin-gonic/gin.(*responseWriter).Flush(0xc0005be700?)
        /app/go-project/ms-gateway/vendor/github.com/gin-gonic/gin/response_writer.go:118 +0x59
net/http/httputil.(*maxLatencyWriter).delayedFlush(0xc0002059e0)
        /usr/local/go/src/net/http/httputil/reverseproxy.go:546 +0x95
created by time.goFunc
        /usr/local/go/src/time/sleep.go:176 +0x32
```

**Comment by wu-sheng** [comment]
fyi @mrproliu

**Comment by akilichen** [comment]
@mrproliu Following this, please help me find the reasons so I can solve the bug. Thanks!

**Reply by mrproliu** [reply]
Sorry, I forget this discussion. Looks like we do not support this scenario. 
If there is dependency the [mux](github.com/gorilla/mux) plugin, for getting the response status code, the agent will wrap the response write in there: https://github.com/apache/skywalking-go/blob/main/plugins/mux/serve_interceptor.go#L64-L74. So I think if you want to resolver this bug, you can enhance in there to let response writer support the `http.Flusher`.

**Reply by akilichen** [reply]
Thanks for your reply. I will try to fix it.
